realm/src/androidTest/java/io/realm/TestHelper.java

```diff
@@ -91,6 +97,73 @@ public static void prepareDatabaseFromAssets(Context context, String realmPath,
         return key;
     }
 
+    /**
+     * Returns a Logger that will fail if it is asked to log a message above a certain level.
+     *
+     * @param failureLevel {@link Log} level from which the unit test will fail.
+     * @return Logger implementation
+     */
+    public static Logger getFailureLogger(final int failureLevel) {
+        return new Logger() {
+
+            private void failIfEqualOrAbove(int logLevel, int failureLevel) {
+                if (logLevel >= failureLevel) {
+                    fail("Message logged that was above valid level: " + logLevel + " >= " + failureLevel);
```

Comment:
Ah you mean CheckedException. That is really not possible from the Logger as that is the standard interface used by the RealmLog. In that case using `fail` is as good as a `RuntimeException`?